src/wordpress/configurator.py

```diff
@@ -18,41 +19,54 @@ class WPRawConfig:
 
     def __init__(self, wp_site):
         self.wp_site = wp_site
+        self._config_infos = None
+        self._user_infos = None
 
     def __repr__(self):
         installed_string = '[ok]' if self.is_installed else '[ko]'
         return "config {0} for {1}".format(installed_string, repr(self.wp_site))
 
-    def run_wp_cli(self, command):
-        # FIXME: maybe we want to bubble up the exception ?
-        try:
-            cmd = "wp {} --path='{}'".format(command, self.wp_site.path)
-            logging.debug("%s - WP CLI %s", self.__class__.__name__, cmd)
-            subprocess.check_output(cmd, stderr=subprocess.STDOUT, shell=True)
-        except subprocess.CalledProcessError as err:
-            logging.error(
-                "%s - WP CLI failed %s - %s - %s",
-                self.__class__.__name__,
-                err, err.returncode, err.output)
-            return False
-
-        # flag out success
-        return True
+    @classmethod
+    def inventory(cls, wp_env, path):
+        # helper function to filter out directories which are part or WP install
+        def keep_wp_sites(dir_name):
+            return dir_name not in WP_DIRS + ENV_DIRS
+
+        # helper class to wrap results
+        WPResult = collections.namedtuple(
+            'WPResult', ['path', 'valid', 'url', 'version', 'db_name', 'db_user', 'admins'])
+
+        # set initial path
+        given_path = os.path.abspath(path)
+        logging.debug('walking through %s', given_path)
+
+        # walk through all subdirs of given path
+        # topdown is true in order modify the dirnames list in-place (and exclude WP dirs)
+        for (parent_path, dir_names, filenames) in os.walk(given_path, topdown=True):
+            # only keep potential WP sites
+            dir_names[:] = [d for d in dir_names if keep_wp_sites(d)]
```

Comment:
dir_names[:] = [d for d in dir_names if keep_wp_sites(d)]

Tu veux que toutes les références au contenu de dir_names soient conservées ?

Reply:
je veux supprimer du dir_names les répertoires worpdress, ainsi que tous leurs sous-repertoires. cette ligne va avec le commentaire sur topdown.

plus clair ?